src/ParsedTwitchMessage.cs

```diff
@@ -29,70 +29,73 @@ public ParsedTwitchMessage(string rawMsg)
 
             // pre-process message, could contain flags that mess with the separators
             string[] components = Regex.Replace(rawMsg, "flags=([^;]*);", "").Split(separator.ToCharArray());
-            
+
             if (components.Length > 2)
             {
                 string tags = components[0];
                 User        = components[1];
                 Message     = string.Join(":", components, 2, components.Length - 2);
-
-                foreach (string str in tags.Split(tagSeparator.ToCharArray()).ToList())
+                if (tags.Length > 0)
```

Comment:
So I wonder if Audica's underlying twitch library returned messages to this class differently than the raw IRC messages.  (or maybe I need to tell Twitch IRC to include more information?)  Here's two examples of raw IRC messages received from Twitch - one from the broadcaster, one from a 'regular' user (my bot account).

```
:steglasaurous!steglasaurous@steglasaurous.tmi.twitch.tv PRIVMSG #steglasaurous :test
:steg_bot!steg_bot@steg_bot.tmi.twitch.tv PRIVMSG #steglasaurous :test twi
```

That's what I'm picking up and sending to ParsedTwitchMessage().  From what I can tell, it seems your code was expecting more information before the first `:` separator.  My simple IRC client doesn't seem to be receiving any of that in the messages, so hence tags was never populated, and this line blew up when it tried to iterate over a null object. :) 

Having said that, this does bring up a good point - effectively anything that was meant for use by mods or the broadcaster would be rendered useless, since it can't tell who has what abilities.  

I can look at the Twitch IRC stuff and see if there's maybe some extra info it needs after connection to include this additional info.

Reply:
I found that apparently you have to request a twitch-specific capability via IRC to include tags in each incoming message.  Now the message is what is expected by ParsedTwitchMessage and works correctly.

Reply:
https://dev.twitch.tv/docs/irc/capabilities/ for more info if you're curious about the specifics :)